pkg/services/ngalert/api/generated_base_api_alertmanager.go

```diff
@@ -18,7 +18,7 @@ import (
 	"github.com/grafana/grafana/pkg/web"
 )
 
-type AlertmanagerApiForkingService interface {
+type AlertmanagerApiHandler interface {
```

Comment:
It feels slightly odd to me that `XApiHandler` interface doesn't have the `handleX` methods hanging off of it. I wonder if the names could be swapped in a couple places? For example, this could simply become `AlertmanagerService` and `AlertmanagerApi` -> `AlertmanagerApiHandler`?

Reply:
I was trying to avoid "Service" because it's such a meaningless name, but I take the point about putting the `handle` functions with the `Handler`. Here's another go:
the generated interface is `AlertmanagerApiServer` and the by-hand implementation is the `AlertmanagerApiHandler`?

Reply:
I haven't looked in-depth at this PR but here to 👍 joe, if possible let's try to avoid the word service as a whole - within Grafana, a service has a completely different meaning to the typical "service" software semantics of grouping a bunch of code that has a single goal.

Reply:
Latest change names things like this:
* The interface is `TypeApi`
* The top-level concrete implementation is `TypeApiHandler`

Other types for the forking alertmanager etc can be named in a way that makes sense. Right now they use `Srv`, but in a separate PR I might try a few names and see what feels good.